aomi/seed.py

```diff
@@ -185,24 +207,49 @@ def app(client, app_obj, opt):
     else:
         name = os.path.splitext(os.path.basename(app_obj['app_file']))[0]
 
-    if not is_tagged(app_obj.get('tags', []), opt.tags):
+    if not is_tagged(opt.tags, app_obj.get('tags', [])):
         log("Skipping %s as it does not have appropriate tags" % name, opt)
         return
 
     app_file = hard_path(app_obj['app_file'], opt.secrets)
+    secret_file(app_file)
     data = yaml.load(open(app_file).read())
-    if 'app_id' not in data \
-       or ('policy' not in data and 'policy_name' not in data):
+
+    ensure_auth(client, 'app-id')
+    if opt.mount_only:
+        log("Only enabling app-id", opt)
+        return
+
+    if 'users' not in data:
         problems("Invalid app file %s" % app_file)
```

Comment:
Why is the file invalid. Maybe follow the pattern of 

Input -> Validator -> Maybe Problems 

again.